### DISCUSSION TITLE
Tamron 18-400mm f/3.5-6.3 Di II VC HLD (B028)

### DISCUSSION THREAD

**Discussion by vector457** [lensfun/lensfun#2355] [General]
Hi, Running Darktable which identifies My (Nikon) Tamron lens as 	Tamron AF 18-400mm f/3.5-6.3 Di II VC HLD
however Lens correction errors/lens not found. The lensfun database does have a 	Tamron 18-400mm f/3.5-6.3 Di II VC HLD (B028) tho. Its just missing the "AF". Which I currently have to manually select from the list.
Whats the best procedure to deal with this to automate it?
thx

**Comment by tuxfanx** [comment]
On my operating systems (Linux and Win10_virtual) darktable 4.8.1 only shows the lens ID 224. This is true for 2 raw images, one taken with a Nikon D5200 and the other with a Nikon D7100 (may be something is wrong with the raws or exiv2).
Automatic lens detection with a raw picture taken with Canon 80D works.
There should be no "AF" in the model name if you are using a current exiv2 installation (see what @kmilos wrote in #2146).
Could you upload a raw picture (taken with lens cap on) [here](https://wilson.bronger.org/calibration)?
Which camera model are you using?
Which operating system are you using?
What is the software version of darktable and exiv2?

**Reply by vector457** [reply]
Hmm this is a bit confusing.
Info I know:
Ubuntu studio 22.04 (haven't updated to 24 yet because google drive application seems to be broken)
darktable 4.8.0
Camera D500 & d5200
exiv2 was not installed ? so not sure how DT was working.
I installed exiv2. (exiv2_0.27.5-3ubuntu1_amd64.deb)
but it made no difference to DT's Tamron lens detection.

_**exiv2 -pt 500_test.nef  |grep 18-400**_   {note:d500 lenscap image}
"Exif.Nikon3.Lens                             Rational    4  18-400mm F3.5-6.3"
"Exif.NikonLd3.LensIDNumber                   Byte        1  Tamron AF 18-400mm F/3.5-6.3 Di II VC HLD"

**_exiv2 -pt 500_7154.NEF |grep 18-400_**  {note:d5200 lenscap image}
"Exif.Nikon3.Lens                             Rational    4  18-400mm F3.5-6.3"
"Exif.NikonLd3.LensIDNumber                   Byte        1  224"

**LensFun**
Now I was under the impression that lensfun db contained all the lenses and their parameters.
**_locate lensfun:  ...found... /usr/share/lensfun/version_1_**
I assume that this is the one to search _/usr/share/lensfun/version_1/**slr-tamron.xml**_
Now I searched that file every which way, there is no "18-400" at all
I tried a few of the other xml files but I still cant find any 18-400
So Im not sure where darktable is finding the non AF 18-400 lens info
Im also not sure how to find which lensfun ver a quick google indicates 
lensfun-update-data
but I get
Command 'lensfun-update-data' not found, but can be installed with:

so yeah a little lost

**Comment by vector457** [comment]
Update.
locate lensfun: ...also found... /snap/darktable/246/usr/share/lensfun/version_1/slr-tamron.xml
which does have a Tamron 18-400mm, no AF

**Comment by kmilos** [comment]
There is always [the exiv2 config file workaround](https://exiv2.org/manpage.html#config_file) at one's disposal. (The "AF" is removed in exiv2 0.28.2, but I imagine it'll take Ubuntu some years to catch up to that version, 24.04 doesn't have it either...)

**Comment by tuxfanx** [comment]
@vector457
If you are using a binary version of darktable (you mentioned snap) then darktable will use the exiv2 data of the exiv2 version it was compiled against (it will then contain its own exiv2 database).
You should use the suggestion described by @kmilos. It seems that your lensfun snap version already contains the lens.

[lensfun-update-data ](https://lensfun.github.io/manual/v0.3.2/lensfun-update-data.html)must be installed separately with the liblensfun-bin package. However, the updated lensfun database will probably not be used by your snap installation. You will need to copy it to the right location (maybe to /home/user/snap/darktable/current/.local/share/) but I am not familiar with Ubuntu and snap, please google it.
